pkg/expression/infer_pushdown.go

```diff
@@ -431,20 +419,56 @@ func IsPushDownEnabled(name string, storeType kv.StoreType) bool {
 
 // PushDownContext is the context used for push down expressions
 type PushDownContext struct {
-	evalCtx  EvalContext
-	client   kv.Client
-	sessVars *variable.SessionVars
+	evalCtx           EvalContext
+	client            kv.Client
+	warnHandler       contextutil.WarnHandler
+	groupConcatMaxLen string
```

Comment:
Why not define it as numeric type?

Reply:
Good suggestion. 
Using a numeric type is much better. I've modified to parse and store `uint64` on sessionvars :+1: